src/Nox.Generator/Application/Dto/EntityUpdateDto.template.cs

```diff
@@ -26,11 +26,20 @@ public partial class {{className}}
     [Required(ErrorMessage = "{{attribute.Name}} is required")]
     {{ end}}
     {{ if componentsInfo[attribute.Name].IsSimpleType -}}
-    public {{componentsInfo[attribute.Name].ComponentType}}{{ if !attribute.IsRequired}}?{{end}} {{attribute.Name}} { get; set; } {{if attribute.IsRequired}}= default!;{{end}}
+    public {{componentsInfo[attribute.Name].ComponentType}}{{ if !attribute.IsRequired}}?{{end}} {{attribute.Name}} { get; set; }{{if attribute.IsRequired}} = default!;{{end}}
     {{- else -}}
-    public {{attribute.Type}}Dto{{- if !attribute.IsRequired}}?{{end}} {{attribute.Name}} { get; set; } {{if attribute.IsRequired}}= default!;{{end}}
+    public {{attribute.Type}}Dto{{- if !attribute.IsRequired}}?{{end}} {{attribute.Name}} { get; set; }{{if attribute.IsRequired}} = default!;{{end}}
     {{- end}}
 {{- end }}
+{{- for relationship in entity.Relationships}}
+    {{- if relationship.Relationship == "ZeroOrOne" || relationship.Relationship == "ExactlyOne" && relationship.ShouldGenerateForeignOnThisSide}}
+
+    /// <summary>
+    /// {{entity.Name}} {{relationship.Description}} {{relationship.Relationship}} {{relationship.EntityPlural}}
+    /// </summary>
+    public string{{if relationship.Relationship == "ZeroOrOne"}}?{{end}} {{relationship.Entity}}Id { get; set; } = null!;
```

Comment:
I don't think that we need to add this attribute since we currently don't have validation on create at all, even for required entity properties. We have validation during partial update in the PartialMapToEntity in mappers. In the same time, we don't have it yet in MapToEntity. I think validation should be implemented on this level for all fields, including navigation properties, but this should be out of the scope of this task.
@rochar should we create a task for this?

Reply:
> The type of the relationship property has to vary based on the entity key primitive type.

Good catch. Actually no, they shouldn't because in our factories almost everywhere we expect a string and have logic to parse it from a string. In this DTO we expect string input from the user and later we convert it into nox type. But I found one small exception, that can be discussed with @rochar later, but doesn't impact the whole flow right now.